controller/src/main/java/org/jboss/as/controller/PersistentResourceXMLDescription.java

```diff
@@ -174,6 +174,7 @@ private void parseGroup(XMLExtendedStreamReader reader, ModelNode op, boolean wi
             if (reader.nextTag() == START_ELEMENT && attributeElements.containsKey(reader.getLocalName())) {
                 AttributeDefinition ad = attributeElements.get(reader.getLocalName());
                 ad.getParser().parseElement(ad, reader, op);
+                ParseUtils.requireNoContent(reader);
```

Comment:
Yes it fixes a problem that new tests found. It was broken in scenario where you had resource that had wrapped properties attribute and than some children. It would fail to read children properly.
Inital bug was using wrong variable for wrapping attribute name when checking for "complete read" in https://github.com/wildfly/wildfly-core/pull/1470/files#diff-429895dde8feaf784ee6dc586357c5bbL101 it worked fine if said attribute was part of attribute group. Most other changes like https://github.com/wildfly/wildfly-core/pull/1470/files#diff-43cfbbd8554a2dc5cbf83398bbb56f83L208 in parser ware just to fix the original bug.
Test that test this is mail-parser.xml https://github.com/wildfly/wildfly-core/pull/1470/files#diff-5f866925622285f598bd12e5bc40920dR21

So in short, when we had wrapped properties/map attribute inside a resource that also has children it would break. 
But if you had same scenario but have said attribute as part of attribute group it would work fine.